src/.clang-format

```diff
@@ -1,50 +1,96 @@
+# -*- mode: yaml; -*-
+
+# Goals
+# - have automated formatting to avoid wasted time
+# - minimize diff-sensitivity to things like changing function names (e.g., don’t want whitespace
+#   diffs on all the parameter lines)
+
+# NB: If clang-format can’t parse this file (or doesn’t understand any of the values, it will format
+#     with the default settings. If this happens it’s important to _revert_ the formatting, not just
+#     reformat after fixing it. This is because some formatting changes aren’t reversible, so you
+#     may introduce formatting regressions if you simply reformat.
+
+# TODO: clang-format 17 adds the ability to include a trailing comment on formatting delimiters, so
+#       once we’re using that version
+#
+#     // NB: why we’re disabling
+#     // clang-format off
+#
+#       can become
+#
+#     // clang-format off: why we’re disabling
+
 AccessModifierOffset: -4
-AlignEscapedNewlinesLeft: true
+AlignAfterOpenBracket: AlwaysBreak
```

Comment:
I prefer [AlignAfterOpenBracket: BAS_BlockIndent](https://clang.llvm.org/docs/ClangFormatStyleOptions.html#alignafteropenbracket).

Reply:
We don't use bitfields enough for it to matter, but I like [AlignConsecutiveBitFields: Consecutive](https://clang.llvm.org/docs/ClangFormatStyleOptions.html#alignconsecutivebitfields).